web/src/views/DandisetLandingView/DandisetDetails.vue

```diff
@@ -152,18 +154,102 @@ export default {
 
       return null;
     },
-    ...mapState('girder', {
-      currentDandiset: (state) => state.currentDandiset,
+    currentVersion() {
+      if (!this.versions) { return draftVersion; }
+      return this.versions[this.currentVersionIndex].version;
+    },
+    currentDandiset() {
+      // Done this way because we'll want to add in
+      // fetching stats from the publish endpoint later on.
+      return this.girderDandiset;
+    },
+    currentVersionIndex() {
```

Comment:
I think this property and `validDandisetVersion` aren't really necessary.
You could simply modify `timelineVersionItemColor` to take the dandiset itself. If it's draft, color=amber, if it's the published dandiset, color=`primary`, otherwise color=grey.
`currentVersion` can be determined by testing if `publishDandiset` is non-null and `publishDandiset.version` is defined.
As I read it, you are currently encoding all the version info into the `versions` list, then decoding it again so it can be displayed.

Reply:
I like this, it cleans up the code a good amount. I'll push a change.